src/providers/r2Provider.ts

```diff
@@ -85,9 +86,33 @@ export class R2Provider implements Provider {
     path: string,
     options?: ReadDirectoryOptions
   ): Promise<ReadDirectoryResult | undefined> {
+    if (path in CACHED_DIRECTORIES) {
+      // @ts-expect-error dates may not be parsed at this point, we take care
+      //  of it below
+      const result: ReadDirectoryResult = CACHED_DIRECTORIES[path];
+
+      for (const file of result.files) {
```

Comment:
Instead of this, can you do this?:

```
const result = JSON.parse(input, function(key, value) {
    if (key == 'date') return new Date(Date.parse(value));
    return value;
});
```

Or something of sort? This speeds up the process and reduces the need of iterating everything again

Reply:
Since `CACHED_DIRECTORIES` is already an object I don't think so?